packages/runner-sdk/lib/sync.ts

```diff
@@ -32,6 +32,8 @@ export abstract class NangoSyncBase extends NangoActionBase {
 
     public abstract batchUpdate<T extends object>(results: T[], model: string): MaybePromise<boolean>;
 
+    public abstract getRecordsByIds<K = any, T = any>(ids: K[], model: string): MaybePromise<Map<K, T>>;
```

Comment:
`K = any`: we actually only support `string | number`

Reply:
Weirdly we don't enforce that on the typescript and `nango.yaml` side for customers, but it does indeed blow up if you try to give it a complex id.